fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java

```diff
@@ -261,6 +261,17 @@ public Void visitCreateMaterializedViewStatement(CreateMaterializedViewStatement
             List<Pair<Column, Integer>> mvColumnPairs = genMaterializedViewColumns(statement);
             List<Column> mvColumns = mvColumnPairs.stream().map(pair -> pair.first).collect(Collectors.toList());
             statement.setMvColumnItems(mvColumns);
+            Map<Integer, Integer> queryToMVOutputIndexMap = Maps.newHashMap();
+            for (int i = 0; i < mvColumnPairs.size(); i++) {
+                Pair<Column, Integer> pair = mvColumnPairs.get(i);
+                queryToMVOutputIndexMap.put(pair.second, i);
+            }
+            List<Integer> queryOutputIndexes = Lists.newArrayList();
+            for (int i = 0; i < mvColumnPairs.size(); i++) {
+                Preconditions.checkState(queryToMVOutputIndexMap.containsKey(i));
+                queryOutputIndexes.add(queryToMVOutputIndexMap.get(i));
```

Comment:
put into a map<int, int> and get from the map, 

is it equivalent to sort the zipWithIndex().sorted(Comparator::comparaing(Pair::getFirst)) ?